### PR TITLE
docs: update Maintainers section

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
In this season finale of reworking the docs: the Maintainers section, mostly. (Previously: #12431, #12505, #14023, #14140, #14152)

### CODE REVIEW

File: docs/Homebrew-brew-Maintainer-Guide.md [hunk 1]
```diff
@@ -10,110 +9,64 @@ Using `gh pr checkout NUMBER` is a super easy way to check out a PR branch using
...
-- `CI / tests (no-compatibility mode)`, `CI / tests (generic OS)` and `CI / tests (Linux)`: These run
-  `brew tests` with various options on Linux.
+- `CI / syntax`: This is run first to check whether the PR passes `brew style` and `brew typecheck`. If this job fails the following jobs will not run.
+- `CI / tap syntax`: This runs `brew style` and `brew audit` on all official taps (note that although this runs on Linux, it does check all cask repos).
+- `CI / docker`: This builds and deploys a new Homebrew Docker image to GitHub Packages and Docker Hub.
+- `CI / test everything (macOS)`: This runs several checks on macOS including `brew tests`, `brew update-tests`, `brew test-bot --only-formulae --test-default-formula`, `brew readall` and `brew doctor`.
+- `CI / tests (no-compatibility mode)`, `CI / tests (generic OS)` and `CI / tests (Linux)`: These run `brew tests` with various options on Linux.
+- `Documentation CI / linting` and `rubydoc`: These check the prose and formatting of the written documentation, and verify the [rubydoc API documentation](https://rubydoc.brew.sh) is built without issue.
```
Comment (Bo98) [inline docs/Homebrew-brew-Maintainer-Guide.md:50]:
```suggestion
- `Documentation CI / linting` and `rubydoc`: These check the prose and formatting of the written documentation, and verify the [rubydoc API documentation](https://rubydoc.brew.sh) can be built without issue.
```

Maybe? The actual website build is performed elsewhere and at a later time.

File: docs/Maintainer-Guidelines.md [hunk 2]
```diff
@@ -23,17 +23,17 @@ Homebrew aims to be the missing package manager for macOS (and Linux). Its prima
 
 1. [Ensure you have set your username and email address properly](https://help.github.com/articles/setting-your-email-in-git/)
 2. Sign off cherry-picks if you amended them (use `git -s`)
-3. If the commit fixes a bug, use “Fixes \#104” syntax to close the bug report and link to the commit
+3. If your commit fixes a bug, use [issue linking syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) (e.g. “Fixes \#104”) to close the bug report and link back to the commit
 
 ### Add comments
 
-It may be enough to refer to an issue ticket, but make sure changes are clear so that if you came to them unaware of the surrounding issues they would make sense to you. Many times on other projects I’ve seen code removed because the new guy didn’t know why it was there. Regressions suck.
+It may be enough to refer to an issue ticket, but make sure changes are clear so that if you came to them unaware of the surrounding issues they would make sense to you. Many times on other projects I’ve seen necessary code removed because the new guy didn’t know why it was there. Regressions suck.
```
Comment (Bo98) [inline docs/Maintainer-Guidelines.md:30]: I know this was already here, but I find the sudden first person sentence here odd.

**Review by Bo98 — APPROVED** [review]

File: docs/Homebrew-brew-Maintainer-Guide.md (hunk 1 above)
Comment (EricFromCanada) [reply docs/Homebrew-brew-Maintainer-Guide.md:50]: True, although it won't actually return a failure until [`--fail-on-warning`](https://github.com/Homebrew/brew/blob/d9ef4545282f0f98664594fb4991b9602a9f27ea/.github/workflows/docs.yml#L67) is enabled (and the failures it produces are fixed) — stay tuned.

File: docs/Maintainer-Guidelines.md (hunk 2 above)
Comment (EricFromCanada) [reply docs/Maintainer-Guidelines.md:30]: No way (for me) to know for sure — this passage pre-dates the [wiki-to-Git transition](https://github.com/Homebrew/brew/commit/19d12aee45ac8d55ecd58a8c072286c84e72041d).

**Pull request merged**